contrib/generate_profile.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+fdbdir=`pwd`/..
+export LD_LIBRARY_PATH=$fdbdir/lib:$LD_LIBRARY_PATH
+export FDB_CLUSTER_FILE=$fdbdir/fdb.cluster
+export LLVM_PROFILE_FILE=$fdbdir/sandbox/fdb-%p.profraw
+../bin/fdbmonitor --conffile ../sandbox/foundationdb.conf --lockfile ../sandbox/fdbmonitor.pid &
+export LLVM_PROFILE_FILE=$fdbdir/sandbox/cli-%m.profraw
+../bin/fdbcli -C ../fdb.cluster --exec 'configure new ssd single'
+export LLVM_PROFILE_FILE=$fdbdir/sandbox/mako-build-%m.profraw
+../bin/mako -p 64 -t 1 --keylen 32 --vallen 16 --mode build --rows 10000  --trace  --trace_format json
+export LLVM_PROFILE_FILE=$fdbdir/sandbox/mako-run-%m.profraw
+../bin/mako -p 1 -t 2 --keylen 32 --vallen 16 --mode run --rows 10000 --transaction grvg7i2gr1:48cr1:48 --seconds 60 --trace ./logs --trace_format json
+
+# Shutdown fdbserver to trigger profile dumping
+pid=$(ps -u $USER | grep fdbserver | fgrep -v grep | awk '{print $2}')
```

Comment:
Make sure that `fdbserver` is exec'ed by the `fdbmonitor` pointed by the `sandbox/fdbmonitor.pid` file content

Reply:
It's doable, however this code assume there is only one fdbserver, if there are multiple fdbserver, it doesn't work.
I'm not sure how Jenkins work, I assume it bring up a VM, and that VM only run my script, so there is only one fdbserver, is that right?

Reply:
There must be only one `fdbserver` process spawn by the `fdbmonitor` process pointed by `fdbmonitor.pid`, no?

Reply:
yes, if so, is it necessary to check which job exec the fdbserver? There is only one fdbmonitor and one fdbserver.

If Jenkins server run multiple build job, will they run in the same VM? If yes, there may be multiple fdbmonitor and multiple fdbserver.

Reply:
In our build pipelines, we can assume that there's only 1 fdbmonitor and 1 fdbserver, but other community members may run this in a shared test environment.

Reply:
if so, I think we need 2 validation:
(1) count numbers of fdbserver, if more than 1, exit 1
(2) find out the parent of that fdbserver, if it's not the right fdbmonitor, exit 1

Is that OK?

Reply:
Ah, no, we don't need to check either.
This script starts its own `fdbmonitor` and it will spawn exactly one `fdbserver`.
All we need to do is to fix this line:
```
pid=$(ps -u $USER | grep fdbserver | fgrep -v grep | awk '{print $2}')
```
such the it should find the right `fdbserver` process with the parent PID same as `sandbox/fdbmonitor.pid`.

Reply:
done